runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java

```diff
@@ -262,6 +285,99 @@ private double getPb(String boss)
 		return personalBest == null ? 0 : personalBest;
 	}
 
+	private void loadPetIcons()
+	{
+		final IndexedSprite[] modIcons = client.getModIcons();
+		if (modIconIdx != -1 || modIcons == null)
+		{
+			return;
+		}
+		final Pet[] pets = getPets();
+		final IndexedSprite[] newModIcons = Arrays.copyOf(modIcons, modIcons.length + pets.length);
+		modIconIdx = modIcons.length;
+
+		for (int i = 0; i < pets.length; i++)
+		{
+			final Pet pet = pets[i];
+
+			try
+			{
+				final BufferedImage image = ImageUtil.resizeImage(itemManager.getImage(pet.getIcon()), 18, 16);
+				final IndexedSprite sprite = ImageUtil.getImageIndexedSprite(image, client);
+				newModIcons[modIconIdx + i] = sprite;
+			}
+			catch (Exception ex)
+			{
+				log.warn("Failed to load the sprite for " + pet.getName(), ex);
+			}
+		}
+
+		log.debug("Adding pet icons");
+		client.setModIcons(newModIcons);
+	}
+
+	/**
+	 * Sets the list of owned pets in local storage for the local player.
+	 *
+	 * @param totalPetList The widget containing the list of owned pets for the local player
+	 */
+	private void setPetList(Widget[] totalPetList)
+	{
+		if (totalPetList == null)
+		{
+			return;
+		}
+
+		List<Pet> petList = new ArrayList<>();
+		for (Widget child : totalPetList)
+		{
+			if (child.getOpacity() == 0)
+			{
+				Pet pet = new Pet(
+					child.getName().replace("<col=ff9040>", "").replace("</col>", ""),
+					child.getItemId());
+				petList.add(pet);
+			}
+		}
+
+		configManager.setRSProfileConfiguration("pets", client.getUsername().toLowerCase(),
```

Comment:
`RSProfileConfiguration` is already per RS-profile, so it is not necessary to set the key to the client username. I would probably just use `chatcommands` as the group and `pets` as the key.